src/common/transformations/src/ov_ops/fully_connected_quantized_legacy.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ov_ops/fully_connected_quantized_legacy.hpp"
+
+#include "matmul_shape_inference.hpp"
+#include "openvino/core/type/element_type.hpp"
+
+namespace ov {
+namespace op {
+namespace internal {
+
+FullyConnectedQuantizedLegacy::FullyConnectedQuantizedLegacy(const OutputVector& arguments,
+                                                             const ov::element::Type output_type)
+    : FullyConnected(OutputVector(arguments.begin(), arguments.begin() + 3), output_type) {
+    for (size_t i = 3; i < arguments.size(); i++) {
+        set_argument(i, arguments[i]);
+    }
+    validate_and_infer_types();
+}
+
+FullyConnectedQuantizedLegacy::FullyConnectedQuantizedLegacy(const ov::Output<Node>& X,
```

Comment:
Do we need these c-tors with explicit args list of we have one with OutputVector already?

Reply:
This is not really necessary, but from my opinion it simplifies transformations code.
This is also related to the possibility of having variable size of the inputs.
For example, this is the code which clones FullyConnectedCompressed GPU version:
``` c++
std::shared_ptr<ov::Node> FullyConnectedCompressed::clone_with_new_inputs(const ov::OutputVector& new_args) const {
    check_new_args_count(this, new_args);

    if (new_args.size() == 4)
        return std::make_shared<FullyConnectedCompressed>(new_args.at(0),
                                                          new_args.at(1),
                                                          new_args.at(2),
                                                          new_args.at(3),
                                                          m_output_type);
    else if (new_args.size() == 5)
        return std::make_shared<FullyConnectedCompressed>(new_args.at(0),
                                                          new_args.at(1),
                                                          new_args.at(2),
                                                          new_args.at(3),
                                                          new_args.at(4),
                                                          m_output_type);
    else if (new_args.size() == 6)
        return std::make_shared<FullyConnectedCompressed>(new_args.at(0),
                                                          new_args.at(1),
                                                          new_args.at(2),
                                                          new_args.at(3),
                                                          new_args.at(4),
                                                          new_args.at(6),
                                                          m_output_type);
    else
        OPENVINO_THROW("Unexpected inputs count for FullyConnectedCompressed op: ", new_args.size());}
```
And this is how it looks with an extra constructor:
``` cpp
std::shared_ptr<ov::Node> FullyConnectedCompressed::clone_with_new_inputs(const ov::OutputVector& new_args) const {
    check_new_args_count(this, new_args);

    return std::make_shared<FullyConnectedCompressed>(new_args, m_output_type);
}
```

Reply:
But to have the second version we need only c-tor with `OutputVector` arg, right? So I'm curious if we can just remove others. From the transformations perspective it can be either simpler (if we just take args from another node and add/modify something. In another cases it's almost same (will need extra braces only)

Reply:
Let's continue in discussion
https://github.com/openvinotoolkit/openvino/pull/26239/files#r1783103394